app/models/user.rb

```diff
@@ -0,0 +1,26 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :comments
+  has_many :likes
```

Comment:
The ERD Diagram you are using is not the one[ in the requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md)

Here is the actual ERD diagram we have inside of the requirements
[ERD Diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png)

Reply:
@wickathou 

Yes, It is in the project requirement. The ERD diagram is a representation of the relationships and associations between tables, surely you cannot expect that to contain Rails conventions, can you?
Please implement the required changes, and focus on the logic of what is being explained. 

At Microverse we teach our students a growth mindset. A growth mindset is the belief that one's abilities and skills can be improved through effort and learning. A fixed mindset is the belief that one's abilities and skills are innate and unchangeable. Having a growth mindset can lead to better outcomes in education, work, and personal development. A growth mindset can help you overcome challenges, embrace feedback, and persist in the face of setbacks. A fixed mindset can limit people's potential, make them avoid criticism, and give up easily when faced with difficulties. Therefore, it is important to cultivate a growth mindset and avoid a fixed mindset in order to achieve one's goals and aspirations.

Kindly focus on learning instead of fighting the people who are trying to help learn the best way to implement solutions. Thank you for your co-operation🙏🏾

Reply:
The ERD diagram we got is this:
![image](https://github.com/wickathou/rails-blog/assets/50066876/bcf827f1-aed9-4ded-90ec-1e964ffb6ab4)

If you see through it, the Likes and Comments user `UserId`, not author ID

I'm only challenging your requests as they are not within the scope of the tasks asked, and there are discrepancies between the requirements and the guide you use for reviews

Reply:
@wickathou 

As I have already mentioned before, focus on the `LOGIC`, in the `User` model you have already added the association with `author_id` as the foreign key with the code ` has_many :posts, foreign_key: :author_id`. Therefore you cannot expect Posts to reference `userId` because it does not exist. It will create errors in your application. If the `author_id` is the foreign key for the `user` and posts, comments, and likes to belong to the `user`. Surely posts, comments, and likes should also referrence `author_id` as a foreign key for the user they belong to.

Note: This is a correction to the ERD diagram. I hope that makes sense.

Reply:
Sure